backend/main.go

```diff
@@ -57,6 +60,36 @@ func main() {
 		}
 		ruleDataList = append(ruleDataList, rd)
 	}
+
+	for {
```

Comment:
Can you write a comment here above the for loop as to why you're waiting for the control status  to be nil?